### PR TITLE
da: try using blobdata for eth fallback

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR enables allows fallback to either 4844 blobdata or calldata based on a flag, `--da.fallback_mode` which can be one of `disabled, calldata, or blobdata`

### CODE REVIEW

**Review by Ferret-san — APPROVED** [review]

File: op-batcher/batcher/driver.go [hunk 1]
```diff
@@ -22,7 +22,14 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
 	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	usableBytesInBlob    = big.NewInt(int64(len(kzg4844.Blob{}) * 31 / 32))
```
Comment (emilianobonassi) [inline op-batcher/batcher/driver.go]: what about to use

https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/blob.go#L16

File: op-batcher/batcher/driver.go [hunk 2]
```diff
@@ -482,6 +489,32 @@ func (l *BatchSubmitter) safeL1Origin(ctx context.Context) (eth.BlockID, error)
...
+		return nil, fmt.Errorf("celestia: fallback disabled")
+	}
+	_, baseFee, blobBaseFee, err := l.Txmgr.SuggestGasPriceCaps(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not get gas price info: %w", err)
+	}
+	blobBaseFee.Mul(blobBaseFee, blobTxBlobGasPerBlob)
+	blobBaseFee.Div(blobBaseFee, usableBytesInBlob)
+	calldataFeePerByte := baseFee.Mul(baseFee, big.NewInt(16))
+	if blobBaseFee.Cmp(calldataFeePerByte) <= 0 {
```
Comment (emilianobonassi) [inline op-batcher/batcher/driver.go]: having a switch based on fee is interesting and probably a feature for upstream optimism

for simplicity i'd fallback only on blob or calldata, not both

calldata or blob fallback controllable by a flag

**Review by emilianobonassi — COMMENTED** [review]
for simplicity i'd fallback only on blob or calldata, not both

calldata or blob fallback controllable by a flag

**Comment by tuxcanfly** [comment]
Agreed @emilianobonassi we can simplify this to fallback to blob/call data based on a feature flag, potentially merging `--da.eth_fallback_disabled` flag with it into an enum.

**Comment by emilianobonassi** [comment]
lgtm

**Comment by tuxcanfly** [comment]
Need to add the fallback flag.

**Comment by tuxcanfly** [comment]
@emilianobonassi good for another review now 👍🏽

**Comment by tuxcanfly** [comment]
This is backward compatible as `da.fallback_mode` defaults to `calldata`, same as current fallback behaviour and the deprecated flag `da.eth_fallback_disabled` now configures `da.fallback_mode=disabled`.

**Pull request merged**